fontforge/parsettf.c

```diff
@@ -4716,30 +4716,43 @@ return;
 	    len = getlong(ttf);
 	    /* language = */ getlong(ttf);
 	}
+	uint16 *dtable = NULL;
+	size_t dlen = len;
+	if (dlen >= 1048576) {
+		// One typeface has over 100,000 entries in a table!
+		IError("Table too large; truncated to 1048576 entries.");
+		dlen = 1048576;
+	}
 	if ( enc->is_unicodebmp && (format==8 || format==10 || format==12))
 	    enc = FindOrMakeEncoding("UnicodeFull");
 
 	if ( format==0 ) {
-	    if ( justinuse==git_normal && map!=NULL && map->enccount<256 ) {
-		map->map = realloc(map->map,256*sizeof(int));
-		memset(map->map,-1,(256-map->enccount)*sizeof(int));
-		map->enccount = map->encmax = 256;
+	    dtable = calloc(dlen, sizeof(uint16));
+	    int encmapminsize = dlen;
+	    if (encmapminsize < 256) encmapminsize = 256;
+	    if ( justinuse==git_normal && map!=NULL && map->enccount<encmapminsize ) {
+		map->map = realloc(map->map,encmapminsize*sizeof(int));
+		memset(map->map,-1,(encmapminsize-map->enccount)*sizeof(int));
+		map->enccount = map->encmax = encmapminsize;
 	    }
+	    int discardc;
 	    for ( i=0; i<len-6; ++i )
-		table[i] = getc(ttf);
+		if (i<dlen-6) dtable[i] = getc(ttf); else discardc = getc(ttf);
```

Comment:
Why not just `else getc(ttf);`? Or if you're worried about unused return (not sure if this works in one line, `else (void)getc(ttc);`

Reply:
I'll give that a try.